tests/php/ORM/SQLSelectTest.php

```diff
@@ -791,6 +792,45 @@ public function testParameterisedLeftJoins()
         $query->execute();
     }
 
+    public function provideUnion()
+    {
+        return [
+            [
+                'unionQuery' => new SQLSelect([1, 2]),
+                'type' => null,
+                'expected' => [
+                    [1 => 1, 2 => 2],
+                ],
+            ],
+            [
+                'unionQuery' => new SQLSelect([1, 2]),
+                'type' => SQLSelect::UNION_DISTINCT,
+                'expected' => [
```

Comment:
Could we update these tests so that `type` `null` and `SQLSelect::UNION_DISTINCT` have different results? Just feed `new SQLSelect` a different array

Reply:
I don't understand why you'd want that (a default union is identical to a DISTINCT union), but I'll do it to avoid ping pong.

Reply:
Done

Reply:
Oh, sorry didn't realise they're identical. Makes sense to just have them test the same thing - maybe just add a comment in the test to say they're different

Er the new $expected isn't what I'd expect, now it doesn't match when `type` is null since it has twice as much expected

Should also add a test to see that an exception is thrown when type is a random string

Reply:
Reverted back to having the default and distinct union tests be identical, and added an appropriate comment.
Added a test for the exception.

> Er the new $expected isn't what I'd expect, now it doesn't match when type is null since it has twice as much expected

Where the union query gives the same results as the base query, only one set of results is returned - because it's DISTINCT. But when the union query gives different results to the base query, those results are included in the result set. This is just SQL working as expected - but it looked strange because of the changed argument passed into the union query. Now that it's reverted, it is more intuitive I think.